tracks/csharp/exercises/leap/mentoring.md

```diff
@@ -20,17 +20,33 @@ using System;
 public static class Leap
 {
     public static bool IsLeapYear(int year)
-        => year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
+        => year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
 }
 ```
 
 ### Common suggestions
 
 - Instead of using several, potentially nested if-statement, suggest to remove the if-statements by returning a single expression using the boolean `&&` and `||` operators.
 - Instead of explicitly returning `true` or `false` (e.g. `if (x) return true else return false`), suggest to simply return the boolean condition (`return x`).
-- If students are doing more than three checks in their if-statement, ask if they could get it down to just three checks. You could give them a hint to following the instructions in the README.
+- If students are doing more than three checks in their if-statement, ask if they could get it down to just three checks. You could give them a hint to following the instructions in the README. There are only two cases that return true:
+  - a year MUST be a multiple of four AND NOT be a multiple of 100
+  - a year MUST be a multiple of 400 (implies multiple of 4 and 100)
+- The _order of operation_ and _evaluation_ matter, see talking points:
```

Comment:
Should this really be a common suggestion? To me, optimizing which checks are executed is nice, but not required at this point in the track (leap being the second exercise). Maybe this section could be moved as a whole to the talking points section?

@robkeim @jpreese what do you think?

Reply:
Substitute this simple exercise with very expensive function calls and suddenly we care. Teach the student to care now so they don't mess up later.

I think that was the original thought behind the note in the ruby and later JavaScript tracks.

Reply:
I think whether or not this should be discussed depends on the student's previous experience.

For someone that is not new to programming but only new to the language it's a good thing to discuss when reviewing the solution.

However, for someone that is just getting started with programming I think that we need to be careful not to overwhelm them. Agreed that order of execution is an important thing to consider, but if you've just completed HelloWorld and this is your second exercise it's not the most important thing for you to be focusing on at that stage IMO.

Reply:
I definitely don't disagree with that. I think two-fer could be between this and hello world.
The issue still is that the experience is different between tracks in a, imo, destructive way.

Would it work if I made these their own topic, but not "forced suggestions"? Instead something the mentor can mention when they approve a solution _without_ parentheses or "wrong" order? I understand that not all tracks need be the same, even though it feels wrong to me.

Reply:
> For someone that is not new to programming but only new to the language it's a good thing to discuss when reviewing the solution.

Well, this is the target audience of Exercism: people that already know how to program but want to become fluent in a language.